src/libs/Animals/Animals.h

```diff
@@ -43,8 +46,8 @@ class ANIMALS : public Entity
     }
 
   private:
-    TSeagulls *seagulls;
-    // TSharks      *sharks;
-    TFishSchools *fishSchools;
-    TButterflies *butterflies;
+    std::shared_ptr<TSeagulls> seagulls;
```

Comment:
Are these copied anywhere? If not, why not use std::unique_ptr instead? It has much less overhead than std::shared_ptr.